### PR TITLE
doc: add documentation for Zigbee scene library

### PR DESCRIPTION
**Pull request by greg-fer**
Added documentation for the new old Zigbee library.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

Related to #5145

### CODE REVIEW

File: include/zigbee/zigbee_zcl_scenes.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+To configure the logging level of the library, use the :option:`CONFIG_ZIGBEE_SCENES_LOG_LEVEL` Kconfig option.
+
+API documentation
+*****************
+
+| Header file: :file:`include/zigbee/zigbee_zcl_scenes.h`
+| Source file: :file:`subsys/zigbee/lib/zigbee_scenes/zigbee_zcl_scenes.c`
+
+.. doxygengroup:: zigbee_logger_ep
```
Comment (greg-fer) [inline include/zigbee/zigbee_zcl_scenes.rst:31]: @tomchy , the header is not updated for doxygen. Can you help?

Comment (tomchy) [reply include/zigbee/zigbee_zcl_scenes.rst]: Fixed!

File: include/zigbee/zigbee_zcl_scenes.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+.. _lib_zigbee_zcl_scenes:
+
+Zigbee ZCL scene helper
+#######################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Zigbee ZCL scene helper library provides a set of functions that implement the callbacks required by the ZCL scene clusters in the application.
```
Comment (tomchy) [inline include/zigbee/zigbee_zcl_scenes.rst:10]: `scene clusters` -> `scene cluster`?

File: include/zigbee/zigbee_zcl_scenes.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+You can use this library to implement these mandatory callbacks and save the configured scenes in the non-volatile memory.
+For this purpose, it uses Zephyr's :ref:`settings_api` subsystem.
+
+This library is capable of recalling attribute values for the following ZCL clusters:
+
+* On/off
+* Level control
+* Window covering
+
+If you are implementing clusters not included in this list, you must implement their logic manually instead of using this library.
```
Comment (tomchy) [inline include/zigbee/zigbee_zcl_scenes.rst:20]: ... `clusters which/that are`?

File: include/zigbee/zigbee_zcl_scenes.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            if (zcl_scenes_cb(bufid) == ZB_FALSE) {
+                  device_cb_param->status = RET_ERROR;
+            }
+            break;
+         }
+      }
+
+Setting the :option:`CONFIG_ZIGBEE_SCENES` option allows you to configure the following library-specific Kconfig options:
+
+* :option:`CONFIG_ZIGBEE_SCENES_ENDPOINT` - This option sets the endpoint number on which the device implements the ZCL scene clusters.
```
Comment (tomchy) [inline include/zigbee/zigbee_zcl_scenes.rst:60]: `clusters` -> `cluster`?

**Review by maciejpietras — APPROVED** [review]

**Review by tomchy — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**